plugin/grpctrace/grpctrace.go

```diff
@@ -39,11 +38,14 @@ type metadataSupplier struct {
 
 func (s *metadataSupplier) Get(key string) string {
 	values := s.metadata.Get(key)
-	return strings.Join(values, ",")
+	if len(values) == 0 {
+		return ""
+	}
+	return values[0]
 }
 
 func (s *metadataSupplier) Set(key string, value string) {
-	s.metadata.Append(key, value)
+	s.metadata.Set(key, value)
```

Comment:
I feel like this needs an explanation. There's no testing here, which suggests the correct behavior might not be well understood. Can you add some detail?

Reply:
This follows the same behavior of the `httptrace`. What happens is if there is an existing `traceparent` metadata in the context (due to an incoming RPC to the server side) and on the server the RPC is forwarded (proxied), the `traceparent` header will be appended, causing it to be invalid. And hence when it is received on another server, the parsing of the `traceparent` header fails and assumes an empty context, since the `traceparent` value is a concatenated with a `,`.

Reply:
I'd say it's a good change, but maybe we should add a text to the `propagation.Supplier` interface that it acts on single-valued keys, so implementations of the interface should avoid appending operations.